src/utils.ts

```diff
@@ -248,7 +254,7 @@ export function executeLegacyOperation(
       const optionsIndex = args.length - 2;
       args[optionsIndex] = Object.assign({}, args[optionsIndex], { session: session });
     } else if (opOptions.session && opOptions.session.hasEnded) {
-      throw new MongoDriverError('Use of expired sessions is not permitted');
+      throw new MongoInvalidArgumentError('Use of expired sessions is not permitted');
```

Comment:
Again, happy to merge this for now but I think this would fall under `MongoExpiredSessionError` which would be under [NODE-3405](https://jira.mongodb.org/browse/NODE-3405).